airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -442,6 +450,9 @@ def _evaluate_direct_relatives() -> Iterator[TIDepStatus]:
                         )
                     )
             elif trigger_rule == TR.ALL_SUCCESS:
+                if ti_needs_expansion and success > 0:
+                    return
```

Comment:
Can we do this outside of the blocks instead of repeating it everywhere? This is both cumbersome and error-prone.